internal/healthchecks/network_check.go

```diff
@@ -32,36 +66,21 @@ func (c NetworkCheck) Name() string {
 }
 
 func (c NetworkCheck) RunCheck(logger *log.Logger) error {
-	// Request to logging API
-	response, err := http.Get(loggingAPIUrl)
-	if err != nil {
-		if isTimeoutError(err) || isConnectionRefusedError(err) {
-			return LogApiConnErr
+	for _, r := range requests {
```

Comment:
Currently `RunCheck` will fail fast on the first encountered failure. Would it make more sense to run all of the checks independently and aggregate the result with `multierr` instead of failing fast?

Reply:
Yes, it should report an aggregate result for the failures, not just fail fast. We (@avilevy18, @braydonk) discussed this at beginning of this task. 

I will address this in the followup PR https://github.com/GoogleCloudPlatform/ops-agent/pull/1139 , since the `API Check` (and now it seems also the `Ports Check`) will benefit from reporting all of the fails/errors. Initially (the current state of #1139) I was only going to separate in different checks (e.g. Log API and Mon API checks), but after discussing the solution was to create a `type MultiHealthCheckError []HealthCheckError` and report each of the relevant failures.